test/e2e/webhook_e2e_test.go

```diff
@@ -0,0 +1,529 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	v1 "github.com/operator-framework/api/pkg/operators/v1"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/install"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
+)
+
+// Global Variables
+const (
+	webhookName = "webhook.test.com"
+)
+
+var _ = Describe("CSVs with a Webhook", func() {
+	var c operatorclient.ClientInterface
+	var crc versioned.Interface
+	var namespace *corev1.Namespace
+	var nsCleanupFunc cleanupFunc
+	BeforeEach(func() {
+		c = newKubeClient(GinkgoT())
+		crc = newCRClient(GinkgoT())
+		namespace, nsCleanupFunc = newNamespace(GinkgoT(), c, genName("webhook-test-"))
+	})
+	AfterEach(func() {
+		nsCleanupFunc()
+	})
+	When("Installed in an OperatorGroup that defines a selector", func() {
+		var nsLabels = map[string]string{
+			"foo": "bar",
+		}
+		var cleanupCSV cleanupFunc
+		var ogSelector = &metav1.LabelSelector{
+			MatchLabels: nsLabels,
+		}
+		var err error
```

Comment:
We only want to declare variables at `Container` level only if we want to share them between multiple specs `It`. `err` isn't something we would want to share across multiple specs. So it is advisable to declare it right before it is used. 
Refer to [this example](https://github.com/operator-framework/operator-lifecycle-manager/blob/db290a7d8503dcfa8f49cb976bcc3da1b054392b/test/e2e/metrics_e2e_test.go#L60)